lib/collection/array.js

```diff
@@ -45,29 +45,25 @@ export class ArrayColl extends KeyValueCollection {
    * to remove them all.
    */
   remove(item) {
-    arrayRemove(this._array, item, false);
-    this._notifyRemoved([item]);
+    if (arrayRemove(this._array, item, false)) {
+      this._notifyRemoved([item]);
+    }
   }
 
   _removeWithoutObserver(item) {
```

Comment:
Could you please document what it now returns?

```suggestion
   /** @returns {integer} number of items removed */
  _removeWithoutObserver(item) {
```